controllers/storagecluster/storagecluster_controller.go

```diff
@@ -144,28 +148,91 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	)
 
-	builder := ctrl.NewControllerManagedBy(mgr).
+	noobaaIgnoreTimeUpdatePredicate := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			if e.ObjectOld == nil || e.ObjectNew == nil {
+				return false
+			}
+			oldObj := e.ObjectOld.(*nbv1.NooBaa)
+			newObj := e.ObjectNew.(*nbv1.NooBaa)
+
+			for _, cond := range oldObj.Status.Conditions {
+				cond.LastHeartbeatTime = metav1.Time{}
+				cond.LastTransitionTime = metav1.Time{}
+			}
+
+			for _, cond := range oldObj.Status.Conditions {
+				cond.LastHeartbeatTime = metav1.Time{}
+				cond.LastTransitionTime = metav1.Time{}
+			}
+
+			oldObj.ObjectMeta.ResourceVersion = ""
+			newObj.ObjectMeta.ResourceVersion = ""
+
+			return !reflect.DeepEqual(oldObj, newObj)
+		},
+	}
+
+	cephClusterIgnoreTimeUpdatePredicate := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			if e.ObjectOld == nil || e.ObjectNew == nil {
+				return false
+			}
+			oldObj := e.ObjectOld.(*cephv1.CephCluster)
+			newObj := e.ObjectNew.(*cephv1.CephCluster)
+
+			for _, cond := range oldObj.Status.Conditions {
+				cond.LastHeartbeatTime = metav1.Time{}
+				cond.LastTransitionTime = metav1.Time{}
+			}
+
+			for _, cond := range oldObj.Status.Conditions {
+				cond.LastHeartbeatTime = metav1.Time{}
+				cond.LastTransitionTime = metav1.Time{}
+			}
+
+			oldObj.ObjectMeta.ResourceVersion = ""
+			newObj.ObjectMeta.ResourceVersion = ""
+
+			return !reflect.DeepEqual(oldObj, newObj)
+		},
+	}
+
+	build := ctrl.NewControllerManagedBy(mgr).
 		For(&ocsv1.StorageCluster{}, builder.WithPredicates(scPredicate)).
-		Owns(&cephv1.CephCluster{}).
-		Owns(&corev1.PersistentVolumeClaim{}, builder.WithPredicates(pvcPredicate)).
+		Owns(&cephv1.CephCluster{}, builder.WithPredicates(cephClusterIgnoreTimeUpdatePredicate)).
+		Owns(&cephv1.CephBlockPool{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Owns(&cephv1.CephFilesystem{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Owns(&cephv1.CephFilesystemSubVolumeGroup{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Owns(&cephv1.CephNFS{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Owns(&cephv1.CephObjectStore{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Owns(&cephv1.CephObjectStoreUser{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Owns(&cephv1.CephRBDMirror{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Owns(&corev1.PersistentVolumeClaim{}, builder.WithPredicates(predicate.GenerationChangedPredicate{}, pvcPredicate)).
```

Comment:
i think this can be removed as well as we don't have any PVC Controller as of now. we removed it long back but we need to see if any other functionality depends on this one before removing it.